mp/src/game/client/momentum/ui/HUD/hud_speedometer.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "hud_speedometer_label.h"
+
+#include "hudelement.h"
+
+#include <vgui_controls/EditablePanel.h>
+#include "mom_player_shared.h"
+
+#define SPEEDOMETER_MAX_LABELS 4
+
+namespace vgui 
+{
+    class Label;
+}
+
+typedef CUtlVector <SpeedometerLabel *> SpeedoLabelList;
+
+class CHudSpeedMeter : public CHudElement, public vgui::EditablePanel
+{
+    DECLARE_CLASS_SIMPLE(CHudSpeedMeter, vgui::EditablePanel);
+
+  public:
+    CHudSpeedMeter(const char *pElementName);
+
+    void OnThink() OVERRIDE;
+    void Init() OVERRIDE;
+    void Reset() OVERRIDE;
+    void FireGameEvent(IGameEvent *pEvent) OVERRIDE;
+    void ApplySchemeSettings(vgui::IScheme *pScheme) OVERRIDE;
+    void PerformLayout() OVERRIDE;
+
+    SpeedometerLabel *GetLabel(int index) { return m_Labels[index]; }
+
+    SpeedoLabelList *GetLabelOrderListPtr() { return &m_LabelOrderList; }
+
+    void ResetLabelOrder();
+
+  private:
+    SpeedometerLabel *m_pAbsSpeedoLabel, *m_pHorizSpeedoLabel, *m_pLastJumpVelLabel, *m_pStageEnterExitVelLabel;
+
+    SpeedoLabelList m_LabelOrderList;
+
+    SpeedometerLabel *m_Labels[SPEEDOMETER_MAX_LABELS] 
+        { m_pAbsSpeedoLabel, m_pHorizSpeedoLabel, m_pLastJumpVelLabel, m_pStageEnterExitVelLabel };
```

Comment:
Let's keep constructing to the constructor.
```suggestion
    SpeedometerLabel *m_Labels[SPEEDOMETER_MAX_LABELS];
```